src/core/segment_buffers/implementations/audio_video/audio_video_segment_buffer.ts

```diff
@@ -376,7 +376,14 @@ export default class AudioVideoSegmentBuffer extends SegmentBuffer {
                       err :
                       new Error("An unknown error occured when doing operations " +
                                 "on the SourceBuffer");
-      this._pendingTask.reject(error);
+      const task = this._pendingTask;
+      if (task.type === SegmentBufferOperation.Push && task.data.length === 0) {
+        if (task.inventoryData !== null) {
```

Comment:
is there any reason to have two `if` conditions instead of one ?

Reply:
you're right they can be combined, will do